src/SFML/Network/IpAddress.cpp

```diff
@@ -210,7 +225,16 @@ bool operator !=(const IpAddress& left, const IpAddress& right)
 ////////////////////////////////////////////////////////////
 bool operator <(const IpAddress& left, const IpAddress& right)
 {
-    return left.toInteger() < right.toInteger();
+    if (left.m_valid < right.m_valid)
+        return true;
+
+    if (right.m_valid < left.m_valid)
+        return false;
+
+    if (left.m_address < right.m_address)
+        return true;
+
+    return false;
 }
```

Comment:
I usually prefer simple and readable code over potential micro-optimizations. And when talking about performance, you should first optimize `operator==` to not call `operator<` twice :wink:
(don't, I'm just saying)